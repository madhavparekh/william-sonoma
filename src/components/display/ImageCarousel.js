import React from 'react';
import PropTypes from 'prop-types';
import { Carousel } from 'react-bootstrap';

class ImageCarousel extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.handleSelect = this.handleSelect.bind(this);

    this.state = {
      index: 0,
      direction: null,
    };
  }

  handleSelect(selectedIndex, e) {
    this.setState({
      index: selectedIndex,
      direction: e.direction,
    });
  }
  render() {
    const { images } = this.props;
    const { index, direction } = this.state;
    return (
      <Carousel
        activeIndex={index}
        direction={direction}
        onSelect={this.handleSelect}
      >
        {images.map((image, indx) => {
          return (
            <Carousel.Item key={indx}>
              <img alt={`img_${indx}`} src={image.href} />
            </Carousel.Item>
          );
        })}
      </Carousel>
    );
  }
}

ImageCarousel.propTypes = {
  images: PropTypes.array,
};

export default ImageCarousel;
