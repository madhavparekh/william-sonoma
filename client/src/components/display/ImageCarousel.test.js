import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import ImageCarousel from './ImageCarousel';

configure({ adapter: new Adapter() });

const images = [
  {
    size: 'm',
    meta: '',
    alt: '',
    rel: 'althero',
    width: 363,
    href:
      'https://www.westelm.com/weimgs/rk/images/wcm/products/201848/0001/organic-luxe-fibrosoft-towels-white-1-m.jpg',
    height: 363,
  },
  {
    size: 'm',
    meta: '',
    alt: '',
    rel: 'althero',
    width: 363,
    href:
      'https://www.westelm.com/weimgs/rk/images/wcm/products/201848/0001/organic-luxe-fibrosoft-towels-midnight-m.jpg',
    height: 363,
  },
  {
    size: 'm',
    meta: '',
    alt: '',
    rel: 'althero',
    width: 363,
    href:
      'https://www.westelm.com/weimgs/rk/images/wcm/products/201848/0001/organic-luxe-fibrosoft-towels-white-m.jpg',
    height: 363,
  },
];

describe('ImageCarousel', () => {
  it('should render correctly', () => {
    const output = shallow(<ImageCarousel images={images} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
