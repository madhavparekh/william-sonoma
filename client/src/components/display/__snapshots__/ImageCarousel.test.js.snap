// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel should render correctly 1`] = `
<Carousel
  activeIndex={0}
  bsClass="carousel"
  controls={true}
  direction={null}
  indicators={true}
  interval={5000}
  nextIcon={
    <Glyphicon
      bsClass="glyphicon"
      glyph="chevron-right"
    />
  }
  nextLabel="Next"
  onSelect={[Function]}
  pauseOnHover={true}
  prevIcon={
    <Glyphicon
      bsClass="glyphicon"
      glyph="chevron-left"
    />
  }
  prevLabel="Previous"
  slide={true}
  wrap={true}
>
  <CarouselItem
    active={false}
    animateIn={false}
    animateOut={false}
    key="0"
  >
    <img
      alt="img_0"
      src="https://www.westelm.com/weimgs/rk/images/wcm/products/201848/0001/organic-luxe-fibrosoft-towels-white-1-m.jpg"
    />
  </CarouselItem>
  <CarouselItem
    active={false}
    animateIn={false}
    animateOut={false}
    key="1"
  >
    <img
      alt="img_1"
      src="https://www.westelm.com/weimgs/rk/images/wcm/products/201848/0001/organic-luxe-fibrosoft-towels-midnight-m.jpg"
    />
  </CarouselItem>
  <CarouselItem
    active={false}
    animateIn={false}
    animateOut={false}
    key="2"
  >
    <img
      alt="img_2"
      src="https://www.westelm.com/weimgs/rk/images/wcm/products/201848/0001/organic-luxe-fibrosoft-towels-white-m.jpg"
    />
  </CarouselItem>
</Carousel>
`;
